pkg/util/deployment/deployment.go

```diff
@@ -130,8 +130,8 @@ func GetPodTemplateSpecHash(template api.PodTemplateSpec) uint32 {
 // Returns the sum of Replicas of the given replication controllers.
 func GetReplicaCountForRCs(replicationControllers []*api.ReplicationController) int {
 	totalReplicaCount := 0
-	for _, rc := range replicationControllers {
-		totalReplicaCount += rc.Spec.Replicas
+	for i := range replicationControllers {
+		totalReplicaCount += replicationControllers[i].Spec.Replicas
```

Comment:
Can you describe some circumstances under which these two forms of the loop are not equivalent? @pweil- suggested that maybe the underlying values could change if you're working with a copy in the former code, but that would imply concurrent access.

Just curious, it caught my attention as a Go puzzler!